lib/galaxy/webapps/base/webapp.py

```diff
@@ -961,12 +964,13 @@ def get_most_recent_history(self):
         if not user:
             return None
         try:
-            recent_history = (
-                self.sa_session.query(self.app.model.History)
+            stmt = (
+                select(self.app.model.History)
                 .filter_by(user=user, deleted=False)
                 .order_by(self.app.model.History.update_time.desc())
-                .first()
+                .limit(1)
             )
+            recent_history = self.sa_session.scalars(stmt).first()
         except NoResultFound:
```

Comment:
Does sqlalchemy still raise NoResultFound and should this better be `one_or_none()` ?

Reply:
nevermind it's at least one ...